File: UTD Battle of the Brains Golang/UTD Battle of the Brains Fall 2023/Bishop Attack/Solution/bishop_attack.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

var file *bufio.Reader = bufio.NewReader(os.Stdin)
var out *bufio.Writer = bufio.NewWriter(os.Stdout)

func printf(f string, a ...interface{}) { fmt.Fprintf(out, f, a...) }
func scanf(f string, a ...interface{})  { fmt.Fscanf(file, f, a...) }

func readLineNumbs(in *bufio.Reader) []string {
	line, _ := in.ReadString('\n')
	line = strings.ReplaceAll(line, "\r", "")
	line = strings.ReplaceAll(line, "\n", "")
	numbs := strings.Split(line, " ")
	return numbs
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	defer out.Flush()
	// Read in input
	line := readLineNumbs(file)
	col := int(line[0][0]-'a') + 1
	row, _ := strconv.Atoi(line[1])

	// Determine the number of squares that can be attacked by a bishop
	// placed at the given position on a standard chessboard.
	topLeft := min(row, col) - 1
	topRight := min(row, 9-col) - 1
	bottomLeft := 8 - max(row, 9-col)
	bottomRight := 8 - max(row, col)

	// Print the Results
	printf("%d\n", topLeft+topRight+bottomLeft+bottomRight+1)
}

File: UTD Battle of the Brains Eclipse Environment/UTD Battle of the Brains Fall 2023 - Judging/src/bishop_attack/BishopAttack.java
package bishop_attack;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.PrintWriter;

public class BishopAttack {
	public static void main(String[] args) throws IOException {
		new BishopAttack().run();
	}
	
	public void run() throws IOException {
		BufferedReader file = new BufferedReader(new InputStreamReader(System.in));
		PrintWriter out = new PrintWriter(System.out);
		
		String[] line = file.readLine().split(" ");
		int col = line[0].charAt(0) - 'a' + 1;
		int row = Integer.parseInt(line[1]);
		file.close();
		
		int topLeft = Math.min(row, col) - 1;
		int topRight = Math.min(row, 9-col) - 1;
		int bottomRight = 8 - Math.max(row, 9-col);
		int bottomLeft = 8 - Math.max(row, col);
		out.println(topLeft + topRight + bottomRight + bottomLeft + 1);
		out.close();
	}
}
